### PR TITLE
dird: fix for crash when starting rescheduled jobs

### PR DESCRIPTION
**Pull request by alaaeddineelamri**
#### Description

This PR fixes a crash that occurs when starting rescheduled jobs.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Is the PR title usable as CHANGELOG entry?
- [x] (will be added by pr-tool) Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems

##### Tests

- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

**Comment by sduehr** [comment]
This occured with the following config:
```
  Reschedule On Error = yes
  Reschedule Interval = 30 minutes
  Reschedule Times = 7
  Max Wait Time = 5 minutes
```
and it occured when the job was started at the rescheduled start time.

So the description should be:
This PR fixes a crash that occurs on starting rescheduled jobs.

File: systemtests/tests/scheduler/testrunner-reschedule [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -o pipefail
```
Comment (pstorz) [inline systemtests/tests/scheduler/testrunner-reschedule:2]:
```suggestion
set -e
set -o pipefail
```

**Review by pstorz — CHANGES_REQUESTED** [review]
See comment, `set -e` is missing in testrunner-reschedule

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**